File: src/main/java/com/gitlab/daring/image/command/parameter/NumberParam.java
package com.gitlab.daring.image.command.parameter;

import static org.apache.commons.lang3.StringUtils.split;

public abstract class NumberParam<T extends Number> extends CommandParam<T> {

    public T minValue;
    public T maxValue;

    public NumberParam(String sv, String sp) {
        super(sv, sp);
        parseSpec();
    }

    protected void parseSpec() {
        String[] ss = split(spec, "-");
        minValue = parseValue(ss[0]);
        maxValue = parseValue(ss[1]);
    }

    public abstract void setNumValue(Number v);

    public double pv() {
        return v.doubleValue() * 0.01;
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/parameter/NumberParam.kt
package com.gitlab.daring.image.command.parameter

abstract class NumberParam<T : Number>(sv: String, sp: String) : CommandParam<T>(sv, sp) {

    @JvmField val minValue: T
    @JvmField val maxValue: T

    init {
        val ss = spec.split('-')
        minValue = parseValue(ss[0])
        maxValue = parseValue(ss[1])
    }

    abstract fun setNumValue(v: Number)

    val pv get() = v.toDouble() * 0.01

}